### DISCUSSION TITLE
Should kwlog.txt be moved to logs?

### DISCUSSION THREAD

**Discussion by skipfire** [jgyates/genmon#1178] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).
- [X] I have included the generator make and model in my submission.

### Version

1.19.04

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

I noticed that kwlog.txt and outage.txt are inside of the configuration folders, with the documentation about using log2ram to reduce writes to the micro-SD card, would it make sense to have these files in /var/log/ or at least somewhere separate from config so that they can also have log2ram covering the frequent writes during an outage?

**Comment by jgyates** [comment]
The idea was to put all of the files related to an installation in one folder to make backup and restore easier. Log files are not restored or backed up by genmon. The default location is /etc/genmon. This folder can be changed to allow [multiple instances of genmon to run at the same time](https://github.com/jgyates/genmon/wiki/Appendix-J-Multiple-Instances-of-Genmon).

Also, Linux has a background process that will rotate and compress log files periodically to prevent the file size form getting to big so having genon store them in this location and expect them to be available with a long history could be problematic.

If you want to add this (RAM disk for power log) functionality you can add a [new folder to log2ram](https://github.com/azlux/log2ram?tab=readme-ov-file#customization) to include /etc/genmon

**Reply by skipfire** [reply]
Thanks for the response. I didn't really want to put the config files in log2ram since those are infrequent writes and ones I would probably not want to lose if log2ram failed to write. I guess a linked file would probably work.

**Reply by jgyates** [reply]
you should be able to just add /etc/genmon to the PATH_DISK variable in /etc/log2ram.conf. The second path should be separated with a semicolon.

**Reply by skipfire** [reply]
But that has the config files in there where config changes could get lost if something happens before the flush to disk.

**Reply by jgyates** [reply, marked as answer]
True.  I think the flush takes place once a day at a minimum or if the system is doing a controlled shutdown. The only case where you could lose data with log2ram is an unexpected power loss. 

This is another reason that I only put non critical info to /var/log. This info is for debugging only.  If you have to have it, write it to the SD card immediately.
